src/config/config.rs

```diff
@@ -13,24 +13,55 @@ use log::warn;
 use std::{
     collections::HashMap,
     fs,
-    path::{Path, PathBuf},
+    path::{Component, Path, PathBuf},
     str::FromStr,
 };
 
+fn canonicalize(path: &Path) -> String {
```

Comment:
Strange that this doesn't exist in `std` in one way or another. But looks like this should work at least for a while until we find a case where it breaks...

Reply:
I love your positive thinking. :rofl:

Reply:
There exists a crate that takes care of this, which might be a bit more complete: https://docs.rs/pathdiff/0.2.0/src/pathdiff/lib.rs.html#34-77

Anyway IMO this function does too many copies that could have been `&Path` slices coalesced into a single owned `PathBuf` in the end, instead of returning a `String` (it's still a path, no?).

Reply:
Performance-wise, it's not the best, however it's done once for all paths at the beginning and that's it. However, we don't use it as a `Path` anyway so there's not much point into storing it as one.